scipy/optimize/_lsq/givens_elimination.pyx

```diff
@@ -44,7 +47,7 @@ def givens_elimination(double [:, ::1] S, double [:] v, double [:] diag):
         if diag[i] == 0:
             continue
 
-        diag_row[:] = 0
+        diag_row[i+1:] = 0
```

Comment:
>in each iteration, so will contain [0, 0, 0, 0.23432, 0, 0, 0] (for i==3). The new code only resets elements i+1:, so (for n==4) could contain [0.11, 0.22, 0.33, 0.23432, 0, 0, 0]

Yes, but [0.11, 0.22, 0.33] won't be used in any way. Does it make sense to you now?